tests/testPupilFactory.py

```diff
@@ -0,0 +1,92 @@
+#
+# LSST Data Management System
+# Copyright 2017 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+from __future__ import absolute_import, division, print_function
+import unittest
+import numpy as np
+
+import lsst.utils.tests
+import lsst.afw.image as afwImage
+import lsst.afw.cameraGeom as afwCameraGeom
+from lsst.afw.geom import degrees
+
+
+class PupilFactoryTestCase(lsst.utils.tests.TestCase):
+    """Test lsst.afw.cameraGeom.PupilFactory"""
+    def setUp(self):
+        self.visitInfo = afwImage.VisitInfo()
+        self.size = 16.8
+        self.npix = 1025  # Odd for reflection test below
+        self.scale = self.size / self.npix
+
+    def testBasePupilFactory(self):
+        pupilFactory = afwCameraGeom.PupilFactory(self.visitInfo, self.size, self.npix)
+        self.assertEqual(pupilFactory.pupilSize, self.size)
+        self.assertEqual(pupilFactory.pupilScale, self.scale)
+        self.assertEqual(pupilFactory.npix, self.npix)
+        with self.assertRaises(NotImplementedError):
+            pupilFactory.getPupil(point=None)
+
+    def testBasePupilFactoryMethods(self):
+        pupilFactory = afwCameraGeom.PupilFactory(self.visitInfo, self.size, self.npix)
+        pupil = pupilFactory._fullPupil()
+        self.assertFloatsEqual(pupil.illuminated, 1.0)
```

Comment:
I was a little surprised that this worked since `illuminated` is a bool array, but it obviously does.